### PR TITLE
[WIP] Feature/pylint-cleanup

### PR DESCRIPTION
**Pull request by fmessmer**
includes #150 and #152

### CODE REVIEW

File: atf_metrics/package.xml [hunk 1]
```diff
@@ -13,6 +13,7 @@
 
     <buildtool_depend>catkin</buildtool_depend>
 
+    <!--exec_depend>atf_core</exec_depend--> <!-- cyclic dependency -->
```
Comment (fmessmer) [inline atf_metrics/package.xml]: these cyclic dependency should be resolved

File: atf_recorder_plugins/package.xml [hunk 2]
```diff
@@ -13,6 +13,7 @@
 
     <buildtool_depend>catkin</buildtool_depend>
 
+    <!--exec_depend>atf_core</exec_depend--> <!-- cyclic dependency -->
```
Comment (fmessmer) [inline atf_recorder_plugins/package.xml]: these cyclic dependency should be resolved

File: atf_test/atf/test_generation_config.yaml [hunk 3]
```diff
@@ -33,12 +33,12 @@ testsuites:                # You have to define at least one testsuite
...
-  - tests:
-      - test1
-    robots:
-      - robot1
-    envs:
-      - env1
-    testblocksets:
-      - testblockset1
-    repetitions: 2
+  # - tests:
```
Comment (floweisshardt) [inline atf_test/atf/test_generation_config.yaml]: @floweisshardt check why ts0 and ts1 cannot be run in parallel

File: atf_recorder_plugins/package.xml (hunk 2 above)
Comment (fmessmer) [reply atf_recorder_plugins/package.xml]: resolved by splitting the `error.py`

File: atf_test/atf/testblocksets/testblockset1.yaml [hunk 4]
```diff
@@ -61,14 +61,14 @@ testblock_small:
       root_frame: world
       measured_frame: reference1
       groundtruth:
-        data: 2.0
-        epsilon: 1.9999
+        data: 3.0
```
Comment (floweisshardt) [inline atf_test/atf/testblocksets/testblockset1.yaml:64]: @floweisshardt verify if there is another tf jump bug?

File: atf_metrics/package.xml (hunk 1 above)
Comment (fmessmer) [reply atf_metrics/package.xml]: resolved by splitting the `error.py`

File: atf_metrics/src/atf_metrics/error.py [hunk 5]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env python
+class ATFAnalyserError(Exception):
```
Comment (floweisshardt) [inline atf_metrics/src/atf_metrics/error.py:2]: @floweisshardt rename to `ATFMetricsError` and `ATFMetricsConfigurationError`

**Comment by floweisshardt** [comment]
@floweisshardt merge with #160 and disable Travis jobs (not required) and enable github actions as required

File: atf_metrics/src/atf_metrics/error.py (hunk 5 above)
Comment (fmessmer) [reply atf_metrics/src/atf_metrics/error.py:2]: I don't know whether `ATFMetricsError` would express the correct meaning if it was `ATFAnalyserError` before...
you will have to check where that error is used and then rename/move/resolve accordingly

**Comment by fmessmer** [comment]
rebased onto github actions

**Comment by fmessmer** [comment]
this PR passed on github actions

File: atf_metrics/src/atf_metrics/error.py (hunk 5 above)
Comment (floweisshardt) [reply atf_metrics/src/atf_metrics/error.py:2]: agreed. let's keep `ATFAnalyserError`

File: atf_test/atf/testblocksets/testblockset1.yaml (hunk 4 above)
Comment (floweisshardt) [reply atf_test/atf/testblocksets/testblockset1.yaml:64]: Running the tests locally I still get a jerk of around `2.0`. @fmessmer Where did you get a higher jerk?

Comment (fmessmer) [reply atf_test/atf/testblocksets/testblockset1.yaml:64]: on some local run - it's been a month ago...

File: .travis.yml [hunk 6]
```diff
@@ -10,14 +10,17 @@ env:
   global:
     - CATKIN_LINT=pedantic
     - NOT_TEST_DOWNSTREAM=true
-    - PARALLEL_TESTS=true
+    - PYLINT_ARGS="--output-format=parseable --errors-only --ignored-modules=tf2_ros"
+    - PYLINT2_CHECK=true
+    - PYLINT3_CHECK=false
+    - PARALLEL_TESTS=false
```
Comment (floweisshardt) [inline .travis.yml:16]: @floweisshardt set to true again

File: atf_test/atf/testblocksets/testblockset1.yaml (hunk 4 above)
Comment (floweisshardt) [reply atf_test/atf/testblocksets/testblockset1.yaml:64]: seems as there are some outliers:
image

this screenshot shows that most of the tests have low jerk but some have higher jerks close to the limit. Needs to be investigated in a new issue:  https://github.com/floweisshardt/atf/issues/165

**Review by floweisshardt — APPROVED** [review]

File: .travis.yml (hunk 6 above)
Comment (floweisshardt) [reply .travis.yml:16]: https://github.com/floweisshardt/atf/issues/166

**Pull request merged**